### PR TITLE
 [FLINK-8007][metrics] Move TestMeter/-Histogram into metrics-core test scope 

### PR DESCRIPTION
**Pull request by zentol**
## What is the purpose of the change

This PR moves the `TestMeter` and `TestingHistogram` to the test sources of `flink-metrics-core`.

The `TestMeter` class was previously part of the main sources of `flink-metrics-core` by accident, while `TestingHistogram` was in the test sources of `flink-runtime`.

The goal consolidate things a bit, and to reduce the dependency on flink-runtime for testing reporters.

Additionally, this PR contains a small cleanup of some reporter poms to remove unused declared / used undeclared dependencies.

## Brief change log

* move TestMeter into test scope of flink-metrics-core
* move TestingHistogram to to flink-metrics-core as
* rename TestingHistogram to TestHistogram

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

File: flink-metrics/flink-metrics-datadog/pom.xml [hunk 1]
```diff
@@ -88,6 +65,7 @@ under the License.
 							<goal>shade</goal>
 						</goals>
 						<configuration>
+							<promoteTransitiveDependencies>true</promoteTransitiveDependencies>
```
Comment (aljoscha) [inline flink-metrics/flink-metrics-datadog/pom.xml]: Why is this needed now?

Comment (zentol) [reply flink-metrics/flink-metrics-datadog/pom.xml]: This is a cleaner approach compared to listing transitive dependencies. We never had a dependency on `okio`; it was only defined to be included in the shaded jar. Promoting transitive dependencies accomplishes the same thing, without the maintainability issues that arise when you bump the version of a dependency.

Comment (aljoscha) [reply flink-metrics/flink-metrics-datadog/pom.xml]: sounds good!

**Comment by zentol** [comment]
merging.
